drainer/syncer.go

```diff
@@ -396,7 +399,16 @@ ForLoop:
 
 	close(fakeBinlogCh)
 	cerr := s.dsyncer.Close()
-	wg.Wait()
+	if cerr != nil {
+		log.Error("Failed to close syncer", zap.Error(cerr))
+	}
+
+	select {
+	case <-wait:
+	case <-time.After(runWaitThreshold):
+		log.Error("Waiting too long for `Syncer.run` to quit.")
+		<-wait
```

Comment:
we can remove it, force it exit

Reply:
OK, do you think 10 seconds is long enough?

Reply:
enough in general